sdk/servicebus/azure-servicebus/azure/servicebus/aio/_servicebus_sender_async.py

```diff
@@ -274,6 +274,9 @@ async def send_messages(self, message):
         if isinstance(message, BatchMessage) and len(message) == 0:  # pylint: disable=len-as-condition
             raise ValueError("A BatchMessage or list of Message must have at least one Message")
 
+        if isinstance(message, (PeekMessage, ReceivedMessageBase)):
```

Comment:
I'm inclined to question whether this check should be part of the helper as well since it's duplicated so many places.  Depending on how you take the unification of the check in BatchMessage from my earlier comment may want to have the helper do all the heavy lifting.  (It can just return the same message if transformation isn't needed; changing the semantics of the helper to something like "_rehydrate_received_message_as_needed" but that's _terrible_ naming so I kinda secretly hope it can move out of BatchMessage and make this an unnecessary suggestion)

This was very rambly so feel free to pick my brain on where my thinking is on this bikeshed.

Reply:
good suggestion ( I was too lazy :P)

moved the transformation part to utils.py to eliminate unnecessary duplication.